app/api_topology.go

```diff
@@ -114,7 +114,11 @@ func handleWebsocket(
 	var (
 		previousTopo render.RenderableNodes
 		tick         = time.Tick(loop)
+		wait         = make(chan struct{}, 1)
 	)
+	rep.WaitOn(wait)
+	defer rep.UnWait(wait)
```

Comment:
Yes I went round and round on this.  I'm not sure either.

I had a version that made a new channel each time and stuck it in a slice on the reporter; the reporter would then close all of them when it received a shortcut report.  The problem is that list would grow unbounded in the absent of the shortcut reports, and we would be continually creating / closing channels.